src/System.Management.Automation/engine/parser/Compiler.cs

```diff
@@ -6300,18 +6306,57 @@ public object VisitMemberExpression(MemberExpressionAst memberExpressionAst)
 
         internal static PSMethodInvocationConstraints GetInvokeMemberConstraints(InvokeMemberExpressionAst invokeMemberExpressionAst)
         {
-            var arguments = invokeMemberExpressionAst.Arguments;
+            Type[] argumentTypes = null;
+            if (invokeMemberExpressionAst.Arguments != null)
+            {
+                argumentTypes = new Type[invokeMemberExpressionAst.Arguments.Count];
+                for (var i = 0; i < invokeMemberExpressionAst.Arguments.Count; i++)
+                {
+                    argumentTypes[i] = GetTypeConstraintForMethodResolution(invokeMemberExpressionAst.Arguments[i]);
+                }
+            }
```

Comment:
I suggest to put `invokeMemberExpressionAst.Arguments` in local variable because it is a compiler and performance is important.

Please use the pattern below too.